File: scala-finch/src/main/scala/org/openapitools/models/RequestTimeFilterPostcodeSectorsDepartureSearch.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import traveltimeplatform._
import java.time.ZonedDateTime
import org.openapitools.models.RequestRangeFull
import org.openapitools.models.RequestTimeFilterPostcodeSectorsProperty
import org.openapitools.models.RequestTransportation
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param transportation 
 * @param travelUnderscoretime 
 * @param departureUnderscoretime 
 * @param reachableUnderscorepostcodesUnderscorethreshold 
 * @param properties 
 * @param range 
 */
case class RequestTimeFilterPostcodeSectorsDepartureSearch(id: String,
                transportation: RequestTransportation,
                travelUnderscoretime: Int,
                departureUnderscoretime: ZonedDateTime,
                reachableUnderscorepostcodesUnderscorethreshold: Double,
                properties: Seq[RequestTimeFilterPostcodeSectorsProperty],
                range: Option[RequestRangeFull]
                )

object RequestTimeFilterPostcodeSectorsDepartureSearch {
    /**
     * Creates the codec for converting RequestTimeFilterPostcodeSectorsDepartureSearch from and to JSON.
     */
    implicit val decoder: Decoder[RequestTimeFilterPostcodeSectorsDepartureSearch] = deriveDecoder
    implicit val encoder: ObjectEncoder[RequestTimeFilterPostcodeSectorsDepartureSearch] = deriveEncoder
}

File: kotlin-spring/src/main/kotlin/traveltimeplatform/models/RequestTimeFilterPostcodeSectorsDepartureSearch.kt
package traveltimeplatform.models

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import traveltimeplatform.models.RequestRangeFull
import traveltimeplatform.models.RequestTimeFilterPostcodeSectorsProperty
import traveltimeplatform.models.RequestTransportation
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * 
 * @param id 
 * @param transportation 
 * @param travelTime 
 * @param departureTime 
 * @param reachablePostcodesThreshold 
 * @param properties 
 * @param range 
 */
data class RequestTimeFilterPostcodeSectorsDepartureSearch (

        @get:NotNull 
        @JsonProperty("id") val id: String,

        @get:NotNull 
        @JsonProperty("transportation") val transportation: RequestTransportation,

        @get:NotNull @get:Min(60) @get:Max(14400) 
        @JsonProperty("travel_time") val travelTime: Int,

        @get:NotNull 
        @JsonProperty("departure_time") val departureTime: java.time.OffsetDateTime,

        @get:NotNull 
        @JsonProperty("reachable_postcodes_threshold") val reachablePostcodesThreshold: Double,

        @get:NotNull 
        @JsonProperty("properties") val properties: List<RequestTimeFilterPostcodeSectorsProperty>,

        @JsonProperty("range") val range: RequestRangeFull? = null
) {

}

